### ISSUE TITLE
[ENH] Add cross-validation for model selection

### ISSUE THREAD

**Issue by mbeyeler** [pulse2percept/pulse2percept#312]
Add cross-validation for the optimizers in the `model_selection` subpackage. There are at least three options:

*  Implement a `crossval_predict` function (like scikit-learn) that takes in an optimizer.
*  Add new objects `ParticleSwarmOptimizerCV` etc. that runs `ParticleSwarmOptimizer` in a cross-validated fashion
*  Add a `cv` argument to the constructor of `ParticleSwarmOptimizer` that triggers cross-validation to happen transparently

Things to consider:
*  The `crossval_predict` approach has the advantage of being familiar to sklearn users. The disadvantage is that the process flow gets kinda involved: `crossval_predict(ParticleSwarmOptimizer(ShapePredictor(AxonMapModel())))`, but maybe that's just how it is.
*  Things can get complicated when we want to support e.g. `groups` splits that use certain columns of `X` to produce the folds

**Comment by mbeyeler** [comment]
we said bye bye
